packages/packages/store/src/__tests__/store.test.tsx

```diff
@@ -0,0 +1,271 @@
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should add a middleware that does not interfere with the state update', () => {
+		// Arrange.
+		addMiddleware( () => ( next: Dispatch<AnyAction> ) => ( action: any ) => {
+			next( action );
+		} );
+
+		const { slice, wrapper } = createStoreEntities();
+
+		// Act.
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 4 );
+	} );
+
+	it( 'should dispatch an action without using hooks', () => {
+		// Arrange.
+		addSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {
+				setValue: ( state, action ) => {
+					state.value = action.payload;
+				},
+			},
+		} );
+
+		const store = createStore();
+
+		// Act.
+		store.dispatch( { type: 'slice/setValue', payload: 6 } );
+
+		const stateResult = store.getState().slice.value;
+
+		// Assert.
+		expect( stateResult ).toBe( 6 );
+	} );
+
+	it( 'should collect actions that are dispatched before the store exist, and run them when the store instance is created', () => {
+		// Act.
+		dispatch( { type: 'slice/setValue', payload: 7 } );
+
+		const { wrapper } = createStoreEntities();
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice.value ), { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 7 );
+	} );
+
+	it( 'should delete the store instance', () => {
+		// Arrange.
+		const { store } = createStoreEntities();
+
+		let instance = getStore();
+
+		// Assert.
+		expect( instance ).toEqual( store );
+
+		// Act.
+		deleteStore();
+
+		instance = getStore();
+
+		// Assert.
+		expect( instance ).toBeNull();
+	} );
+
+	it( 'should delete the added slices', () => {
+		// Arrange.
+		jest.spyOn( console, 'error' ).mockImplementation( () => {} );
+
+		createStoreEntities();
+
+		// Act.
+		deleteStore();
+
+		// Arrange.
+		const store = createStore();
+
+		const wrapper: React.FC = ( { children } ) => (
+			<StoreProvider store={ store }>
+				{ children }
+			</StoreProvider>
+		);
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice ), { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBeUndefined();
+
+		// Redux sends an error to the console when no reducer is found.
+		// eslint-disable-next-line no-console
+		expect( console.error ).toHaveBeenCalled();
+	} );
+
+	it( 'should delete the added middlewares', () => {
+		// Arrange.
+		// Registering a middleware that blocks the state update by not running next(action).
+		addMiddleware( () => () => () => {} );
```

Comment:
you can add a mock method inside the middleware and make sure it doesn't get called, I think it'll be clearer than writing a comment

Reply:
The function inside the middleware is always called, but blocks the state by no running next(action).